iso15118/secc/controller/ev_data.py

```diff
@@ -139,36 +124,398 @@ def __init__(
+            if control_mode == ControlMode.SCHEDULED:
+                params = dc_charge_loop_req.bpt_scheduled_params
+                # Target Current and Voltage only in Scheduled
+                self.target_current = params.ev_target_current.get_decimal_value()
+                self.target_voltage = params.ev_target_voltage.get_decimal_value()
+                _update_common_bpt_limits(dc_limits, params)
+            elif control_mode == ControlMode.DYNAMIC:
+                params = dc_charge_loop_req.bpt_dynamic_params
+                # Departure time only in Dynamic
+                if params.departure_time:
+                    self.departure_time = params.departure_time
+                # V2X Limits only in Dynamic BPT
+                if params.ev_max_v2x_energy_request:
+                    self.max_v2x_energy_request = params.ev_max_v2x_energy_request.get_decimal_value() # noqa: E501
+                if params.ev_min_v2x_energy_request:
+                    self.min_v2x_energy_request = params.ev_min_v2x_energy_request.get_decimal_value() # noqa: E501
+                _update_common_bpt_limits(dc_limits, params) 
+        else:
+            return
+
+        def _update_common_limits(
+            dc_limits: EVDCCLLimits,
+            params: Union[ScheduledDCChargeLoopReqParams,
+                            DynamicDCChargeLoopReqParams,
+                            BPTScheduledDCChargeLoopReqParams,
+                            BPTDynamicDCChargeLoopReqParams]
+        ):
+            if params.ev_max_charge_power:
+                dc_limits.max_charge_power = params.ev_max_charge_power.get_decimal_value() # noqa: E501
+            if params.ev_min_charge_power:
+                dc_limits.min_charge_power = params.ev_min_charge_power.get_decimal_value() # noqa: E501
+            if params.ev_max_charge_current:
+                dc_limits.max_charge_current = params.ev_max_charge_current.get_decimal_value()  # noqa: E501
+            if params.ev_max_voltage:
+                dc_limits.max_voltage = params.ev_max_voltage.get_decimal_value()  # noqa: E501
+            if params.ev_min_voltage:
+                dc_limits.min_voltage = params.ev_min_voltage.get_decimal_value()  # noqa: E501
+
+            if params.ev_target_energy_request:
+                self.target_energy_request = params.ev_target_energy_request.get_decimal_value()
+            if params.ev_max_energy_request:
+                self.maximum_energy_request = params.ev_max_energy_request.get_decimal_value()
+            if params.ev_min_energy_request:
+                self.minimum_energy_request = params.ev_min_energy_request.get_decimal_value()
+
+        def _update_common_bpt_limits(
+                dc_limits: EVDCCLLimits,
+                params: Union[BPTScheduledDCChargeLoopReqParams,
+                            BPTDynamicDCChargeLoopReqParams]
+        ):
+            _update_common_limits(dc_limits, params)
```

Comment:
Same comment as above - would member method be better? You are leaving the underscore to indicate it is supposed to be private anyway...